### PR TITLE
Fix usenet crawler searchstring paramaters

### PR DESCRIPTION
**Pull request by KontiSR**
Fix usenet crawler searchstring paramaters 
'ep' and 'rid' don't work anymore. I've replaced these with 'episode' and 'tvrage' specific for usenet-crawler. Searching series/anime should work now, while other newznab based providers should not be effected.

### CODE REVIEW

**Comment by fernandog** [comment]
what about tvdb?

File: sickbeard/providers/newznab.py [hunk 1]
```diff
@@ -141,9 +141,11 @@ def _get_season_search_strings(self, ep_obj):
 
         # search
         rid = helpers.mapIndexersToShow(ep_obj.show)[2]
+        #added exception for usenet-crawler, as it seems they have changed something within their api
+        tvrage = 'rid' if self.name <> "Usenet-Crawler" else 'tvrage'
```
Comment (miigotu) [inline sickbeard/providers/newznab.py]: why `<>` and not in ?

Comment (KontiSR) [inline sickbeard/providers/newznab.py]: I'm just doing to many different languages at the moment. <> works the same as !=. <> is supported, but i just read it's deprecated. If you'd like I can change it to !=.

Personally i only use "not in", with lists or tuples. But also that's possible, if you prefer, I can also use that.

**Comment by fernandog** [comment]
@KontiSR merge conflicts

Step 1: From your project repository, check out a new branch and test the changes.
git checkout -b KontiSR-dev-fix-usenet-crawler-searchstring-para develop
git pull git://github.com/KontiSR/SickRage.git dev-fix-usenet-crawler-searchstring-para

Step 2: Merge the changes and update on GitHub.
git checkout develop
git merge --no-ff KontiSR-dev-fix-usenet-crawler-searchstring-para
git push origin develop

File: sickbeard/providers/newznab.py [hunk 2]
```diff
@@ -162,22 +164,25 @@ def _get_episode_search_strings(self, ep_obj, add_string=''):
 
         if not ep_obj:
             return [params]
+        
+        episode = 'ep' if self.name <> "Usenet-Crawler" else 'episode'
```
Comment (autronix) [inline sickbeard/providers/newznab.py]: This is not unique to Usenet-Crawler, I encounter the same behavior wih nzb.su; could possibly apply to others - maybe looping through a list of providers would be a better choice than setting a single provider string

File: sickbeard/providers/newznab.py [hunk 3]
```diff
@@ -195,11 +200,11 @@ def _get_episode_search_strings(self, ep_obj, add_string=''):
                 params['q'] = helpers.sanitizeSceneName(cur_exception)
                 paramsNoEp = params.copy()
                 
-                paramsNoEp['q'] = paramsNoEp['q'] + " " + str(paramsNoEp['ep'])
-                if "ep" in paramsNoEp:
-                    paramsNoEp.pop("ep")
+                paramsNoEp['q'] = paramsNoEp['q'] + " " + str(paramsNoEp[episode])
```
Comment (autronix) [inline sickbeard/providers/newznab.py]: My understanding is that paramsNoEp['episode'] is already a string formated by "%i" % ... in line 171. Also worth noting that for both providers I've tried (usenet-crawler and nzb.su) having an episode without a 0 padding when when in single digits does not yield a result: q=Show.Name+3 => no result while q=Show.Name+03 has result

Comment (miigotu) [inline sickbeard/providers/newznab.py]: There should be no ep # in the q string if its added as ep/episode parameter. Thats why there is params and paramsNoEp

File: sickbeard/providers/newznab.py (hunk 1 above)
Comment (miigotu) [inline sickbeard/providers/newznab.py]: To be pyhtoninc you should only use != (cstyle comparison operators) when comparing integers

I would rather see 
```python
tvrage = 'rid' if self.name not in "Usenet-Crawler" else 'tvrage' 
```
or
```python
tvrage = ('rid', 'tvrage') ["Usenet-Crawler" in self.name] 
```

Comment (KontiSR) [inline sickbeard/providers/newznab.py]: Aaargggg, you'r right. Crap. okay, i'm just going to pop the paramater

**Comment by KontiSR** [comment]
Aah it's the same for ep as it is for rid. Both paramaters are not functioning. Leaving them out has the same effect as adding the 'episode' and 'tvrage'. Pff such a mess. If I also leave episode out, sickrage will get back an enormous set of results to parse, each time it's doing a daily/hourly/manual search.

I'm really in doubt if we should want to fix this in SR.

**Comment by autronix** [comment]
Same behavior appears on nzb.su, not only Usenet-Crawler.
It might be ideal to have a list rather than just a string matching Usenet-Crawler

File: sickbeard/providers/newznab.py [hunk 4]
```diff
@@ -147,7 +147,7 @@ def _get_season_search_strings(self, ep_obj):
 
         # search
         rid = helpers.mapIndexersToShow(ep_obj.show)[2]
-        if rid:
+        if rid and self.name not in "Usenet-Crawler":
```
Comment (autronix) [inline sickbeard/providers/newznab.py]: We would need to have something like 
```
newznab_providers = ["Usenet-Crawler", "NZB.su", ....]
if rid and self.name not in newznab_providers:
[.....]

```

File: sickbeard/providers/newznab.py [hunk 5]
```diff
@@ -202,9 +203,9 @@ def _get_episode_search_strings(self, ep_obj, add_string=''):
 
             if ep_obj.show.anime:
                 paramsNoEp = params.copy()
-                paramsNoEp['q'] = paramsNoEp['q'] + " " + paramsNoEp['ep']
-                if "ep" in paramsNoEp:
-                    paramsNoEp.pop("ep")
+                paramsNoEp['q'] = paramsNoEp['q'] + " " + paramsNoEp[episode]
```
Comment (autronix) [inline sickbeard/providers/newznab.py]:
```python
if int(paramsNoEp[episode]) < 10:
    paramsNoEp['q'] = paramsNoEp['q'] + " 0" + paramsNoEp[episode]
else:
    paramsNoEp['q'] = paramsNoEp['q'] + " " + paramsNoEp[episode]
```

**Comment by thezoggy** [comment]
you dont do rid + query. you do one or the other.... notice that this 'issue' doesnt exist in sonarr/sb

**Comment by miigotu** [comment]
Calm down @thezoggy I'm working on it now.

**Comment by KontiSR** [comment]
We've already realized that the way it has been implemented is nog ideal. Im also werking on a rewrite, where i dump the unnecessary paramaters all together for newznab.

**Comment by miigotu** [comment]
@KontiSR I have it half way there already. Just need to decide if I should use season/ep params with the q string and no rid, or if I should put the ep/season in the q string.

Debating just using season, ep, and rid and skipping scene name exceptions for newznab (no q searches)

I promise if I do that, and they still have some complaint, then they are full of shit. It's one thing that I take care of everything they point out and then some every time, but I don't like how it's made out to be that someone is an idiot or something. Like seriously, even SB is far from perfect.

**Comment by thezoggy** [comment]
@miigotu i'm plenty calm, i dont use sr :)

File: sickbeard/providers/newznab.py (hunk 5 above)
Comment (thezoggy) [inline sickbeard/providers/newznab.py]: http://stackoverflow.com/questions/339007/nicest-way-to-pad-zeroes-to-string

Comment (autronix) [inline sickbeard/providers/newznab.py]: @thezoggy thanks - I haven't done python in a very long time
```python
paramsNoEp['q'] = paramsNoEp['q'] + " " + paramsNoEp[episode].zfill(2)
```

**Comment by Taloth** [comment]
@miigotu I was warned of possible issues with Usenet-crawler by zoggy, so I tested it a bit yesterday on Better Call Saul. The api seems to be working like other nn indexers.
Basically you should do `rid+season+ep` or `q+season+ep` depending on whether you have the tvrageid yourself. IIRC for SB they did the `q` search only if either the tvrageid is 0 or no results were found.
The problem with `rid+q+season+ep` is that, if the indexer doesn't have the tvrageid associated with any release, you'll get no results at all, and if there are associated releases, the `q` parameter will simple filter further instead of adding additional releases by scene exception. To put it simple, all those parameters use `and` operators.

I hope that helps. Feel free to bug me if you have questions.

**Comment by KontiSR** [comment]
@miigotu Tnx for giving this some attention! I think we all have a pretty good understanding on how to get this fixed. As this is not really the ideal place to discus this (we already have have the open issue), im going to close this one.

**Pull request closed without merging**